circle.yml

```diff
@@ -6,13 +6,7 @@ test:
     - NODE_ENV=test make client/config/index.js:
         parallel: true
   override:
-    - bin/run-lint :
-        parallel: true
-        files:
-          - client/**/*.js
-          - client/**/*.jsx
-          - server/**/*.js
-          - server/**/*.jsx
+    - bin/run-lint $(git diff --name-only $(git merge-base $(git rev-parse --abbrev-ref HEAD) origin/master)..HEAD *js *jsx)
```

Comment:
> What if there are errors in the other files?

You mean JS errors, not rule-breaking, right? This approach moves us from linting the whole codebase each time to lint the specific diff's we want to merge. Granted that all changes go through this cycle of building first a PR, we should not have undetected errors.

One concern I have after reading @blowery question above about master merges, though, is that we might need also to execute run-lint script on that cases. That is something we can do to cover all "entry points" to the codebase - if CircleCI also run on master commits. I am on mobile and I cannot check that. Would be great to learn more about how calypso uses CircleCI: does it send reports on error to some places (slack, email, etc)?

>what if we changed the eslint configuration or we upgraded ESLint or in some other way modified how the linter works?

As long as ESLint keeps accepting a list of files to check we are safe. Right now, I cannot see a way how this PR would be affected by the things you suggest - or any other changes of that kind, to that matter. I would love to be illustrated if I am missing something obvious. The point is, nevertheless, very relevant for the next step -linting only specific lines- as we are going to depend more on ESLint specifics. Thanks for the food for thought.